src/string_util.cc

```diff
@@ -16,11 +15,11 @@ namespace benchmark {
 namespace {
 
 // kilo, Mega, Giga, Tera, Peta, Exa, Zetta, Yotta.
-const char kBigSIUnits[] = "kMGTPEZY";
+const std::string kBigSIUnits[] = {"k", "M", "G", "T", "P", "E", "Z", "Y"};
```

Comment:
unfortunately, this causes potential issues: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

this will need to be

`const char* const kBigSIUnits[] = {"k", "M", ...};`

to avoid any issues with non-trivial destructors.

Reply:
I corrected this and also removed the [if-clause](https://github.com/google/benchmark/blob/56898e9a92fba537671d5462df9c5ef2ea6a823a/src/string_util.cc#L97) from this line, as we now already have the “i” at the end of each prefix of the IEC units.

```cpp
std::string ExponentToPrefix(int64_t exponent, bool iec) {
  if (exponent == 0) return "";

  const int64_t index = (exponent > 0 ? exponent - 1 : -exponent - 1);
  if (index >= kUnitsSize) return "";

 const char* const* array =
      (exponent > 0 ? (iec ? kBigIECUnits : kBigSIUnits) : kSmallSIUnits);

  return std::string(array[index]);
}
```

Kindly review this modified function.